src/suricata.c

```diff
@@ -345,6 +345,77 @@ void GlobalInits()
     SupportFastPatternForSigMatchTypes();
 }
 
+void GlobalsDestroy(SCInstance *suri)
+{
+    HostShutdown();
+    HTPFreeConfig();
+    HTPAtExitPrintStats();
+
+#ifdef DBG_MEM_ALLOC
+    SCLogInfo("Total memory used (without SCFree()): %"PRIdMAX, (intmax_t)global_mem);
+#ifdef DBG_MEM_ALLOC_SKIP_STARTUP
+    print_mem_flag = 0;
+#endif
+#endif
+
+    SCPidfileRemove(suri->pid_filename);
```

Comment:
We could start a other suri if PID is removed. This could result in two suricata logging to file at the same time in same file. Could we move that later ?